sys/kern/syscalls.c

```diff
@@ -531,13 +531,23 @@ static int sys_clock_nanosleep(proc_t *p, clock_nanosleep_args_t *args,
   clockid_t clock_id = args->clock_id;
   int flags = args->flags;
   const timespec_t *u_rqtp = args->rqtp;
-  timespec_t rqtp;
-  int error;
+  timespec_t *u_rmtp = args->rmtp;
+  timespec_t rqtp, rmtp;
+  int error, error2;
 
-  if ((error = copyin_s(u_rqtp, rqtp)))
-    return error;
+  if ((error = copyin_s(u_rqtp, rqtp))) {
+    *res = error;
+    return 0;
+  }
 
-  return do_clock_nanosleep(clock_id, flags, &rqtp, NULL);
+  error = do_clock_nanosleep(clock_id, flags, &rqtp, u_rmtp ? &rmtp : NULL);
+
+  if ((u_rmtp != NULL && (error == 0 || error == EINTR)) &&
+      (flags & TIMER_ABSTIME) == 0 && (error2 = copyout_s(rmtp, u_rmtp)))
+    error = error2;
+
+  *res = error;
```

Comment:
You could move `*res = error` before if statement then `error2` is not necessary.

Reply:
Done.